### PR TITLE
Kindergeldübertrag

### LINKED ISSUE #742: BUG: Kindergeldübertrag
### Bug description

The current version of GETTSIM doesn't support the Kindergeldübertrag.

The Kindergeldübertrag is relevant whenever children are not part of the Bedarfsgemeinschaft of their parents (e.g. due to enough income through Unterhalt). Then, the Kindergeld that is not necessary to cover the needs of the child are considered as parental income for Bürgergeld.

### PR DESCRIPTION
**Pull request by MImmesberger**
### What problem do you want to solve?

Closes #742 

The implementation will still rely on manually removing children from their Bedarfsgemeinschaft (via `eigenbedarf_gedeckt`) until the corresponding issue #622 has been addressed.

### CODE REVIEW

File: src/_gettsim/transfers/kindergeld.py [hunk 1]
```diff
@@ -174,3 +181,104 @@ def kind_bis_10_mit_kindergeld(
...
     return out
+
+
+def _kindergeld_per_child_m(
+    kindergeld_m: float,
+    kindergeld_anz_ansprüche: int,
+) -> float:
+    """Kindergeld per child.
+
+    Helper function for `kindergeld_zur_bedarfdeckung_m`. Returns the average Kindergeld
```
Comment (ChristianZimpelmann) [inline src/_gettsim/transfers/kindergeld.py:192]: Would revert the sentences: first what it calculates. Then (potentially) why/where we need it.

File: src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py [hunk 2]
```diff
@@ -44,6 +44,38 @@ def arbeitsl_geld_2_eink_m(  # noqa: PLR0913
     )
 
 
+def _arbeitsl_geld_2_eink_ohne_kindergeldübertrag_m(
```
Comment (ChristianZimpelmann) [inline src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py:47]: Somewhat confusing that `_arbeitsl_geld_2_eink_ohne_kindergeldübertrag_m` contains more components than `arbeitsl_geld_2_eink_m`

At some point we need to rename one of the functions and/or refactor the module.

File: src/_gettsim/transfers/kindergeld.py [hunk 3]
```diff
@@ -174,3 +181,104 @@ def kind_bis_10_mit_kindergeld(
...
+    # because the child doesn't drop out of Bedarfsgemeinschaft endogenously).
+    # https://github.com/iza-institute-of-labor-economics/gettsim/issues/622
+    fehlbetrag = max(
+        arbeitsl_geld_2_regelbedarf_m_bg
+        - _arbeitsl_geld_2_eink_ohne_kindergeldübertrag_m_bg,
+        0.0,
+    )
+    if (
+        not eigenbedarf_gedeckt
+    ):  # Same Bedarfsgemeinschaft as parents or Bedarf is not covered
```
Comment (ChristianZimpelmann) [inline src/_gettsim/transfers/kindergeld.py:278]: split up comment to this if and the next elif statement

File: src/_gettsim/transfers/kindergeld.py [hunk 4]
```diff
@@ -174,3 +181,104 @@ def kind_bis_10_mit_kindergeld(
...
+    """
+    # TODO (@MImmesberger): Remove `eigenbedarf_gedeckt` conditions once
+    # Bedarfsgemeinschaft is fully endogenous. This is a temporary fix. Without it,
+    # Kindergeld would be counted twice as income of the Bedarfsgemeinschaft (one time
+    # the full amount for the child and one time the Kindergeldübertrag for the parent -
+    # because the child doesn't drop out of Bedarfsgemeinschaft endogenously).
+    # https://github.com/iza-institute-of-labor-economics/gettsim/issues/622
+    fehlbetrag = max(
+        arbeitsl_geld_2_regelbedarf_m_bg
+        - _arbeitsl_geld_2_eink_ohne_kindergeldübertrag_m_bg,
```
Comment (ChristianZimpelmann) [inline src/_gettsim/transfers/kindergeld.py:273]: Add ToDo that most likely also (potential) Kinderwohngeld is considered here.

File: src/_gettsim/transfers/kindergeld.py [hunk 5]
```diff
@@ -13,12 +15,17 @@
         "source_col": "kindergeld_anspruch",
         "aggr": "sum",
     },
+    "kindergeld_übertrag_m": {
```
Comment (ChristianZimpelmann) [inline src/_gettsim/transfers/kindergeld.py:18]: The additions in this module do not really fit here. How about a new module in the `arbeitsl_geld_2` directory?

**Review by ChristianZimpelmann — COMMENTED** [review]
Thanks! Seems overall nicely done. But a couple of comments.

File: src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py (hunk 2 above)
Comment (MImmesberger) [reply src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py:47]: I agree, that's very confusing.

I renamed it in the following way:
`_arbeitsl_geld_2_eink_ohne_kindergeldübertrag_m` -> `arbeitsl_geld_2_eink_m` (=SGB 2 income without Kindergeldübertrag
`arbeitsl_geld_2_eink_m` -> `arbeitsl_geld_2_nettoeink_ohne_transfers_m` (Employment income, capital gains, etc.)

Do you think that's better?

File: src/_gettsim/transfers/kindergeld.py (hunk 5 above)
Comment (MImmesberger) [reply src/_gettsim/transfers/kindergeld.py:18]: Good idea!

File: src/_gettsim/transfers/arbeitsl_geld_2/kindergelduebertrag.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
+"""Module for the calculation of the Kindergeldübertrag."""
```
Comment (MImmesberger) [inline src/_gettsim/transfers/arbeitsl_geld_2/kindergelduebertrag.py:1]: For some reason, using the correct Umlaut in the filename results in gibberish and I get a ruff error in the pre-commits.

**Comment by MImmesberger** [comment]
Should be more careful with squash commits when working on several PRs at the same time, somehow I broke the commit history. Will fix that tomorrow or do a new PR with cherry-picked commits.

File: src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py (hunk 2 above)
Comment (ChristianZimpelmann) [reply src/_gettsim/transfers/arbeitsl_geld_2/arbeitsl_geld_2_eink.py]: Sounds good!

**Comment by MImmesberger** [comment]
I have cherry-picked the relevant commits and included them in #751. Closing this one.

**Pull request closed without merging**